lib/segment/src/types.rs

```diff
@@ -233,6 +234,89 @@ impl PartialEq for ScoredPoint {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub enum Score {
+    Id(PointIdType),
+    Similarity(ScoreType),
+    IntValue(IntPayloadType),
+    FloatValue(FloatPayloadType),
```

Comment:
I am not sure I agree that Id and Similarity should be separate. 
In my understanding any score should be possible to represent as a number. Even if we order by time, the score value should be timestamp as a number.

We can consider enum of i64 and float though, cause float64 by itself not always have enough capacity.